src/Symfony/Component/DependencyInjection/ChildDefinition.php

```diff
@@ -26,7 +26,7 @@ class ChildDefinition extends Definition
     private $changes = array();
 
     /**
-     * @param string $parent The id of Definition instance to decorate
+     * @param string|null $parent The id of Definition instance to decorate
```

Comment:
I don't like the idea of allowing null here. A ChildDefinition registered in the container must not have a null parent id (it would break things).

If you use a ChildDefinition temporarily to resolve instanceof configurations in the loaders, I would rather use a dummy parent id (which will not go outside the loader anyway)